validator/api/src/main/java/tech/pegasys/teku/validator/api/FileBackedGraffitiProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+
+/**
+ * A type to hold the current graffiti value. Graffiti can be loaded from 2 places:
+ *
+ * <ul>
+ *   <li>The default value supplied at initialisation
+ *   <li>The graffiti file which will be regularly checked
+ * </ul>
+ *
+ * In the case that the file cannot be read, the default graffiti value will be used.
+ */
+public class FileBackedGraffitiProvider implements GraffitiProvider {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Optional<Bytes32> defaultGraffiti;
+  private final Optional<Path> graffitiFile;
+
+  public FileBackedGraffitiProvider(
```

Comment:
nit: could consider creating a default constructor 
```
public FileBackedGraffitiProvider() {
this(Optional.empty(), Optional.empty());
}
```
nearly half the calls to this provider are empty, so it'd clean up some of that...

as an aside, generally when we say 'nit' we're just nit-picking and its optional - just in case that's not something you generally see...

Reply:
Ha - thanks. I haven't seen nit before. In general as long as I agree with a nit I'll strive to implement it.

Reply:
For completeness, type erasure prevents the introduction of these constructors

```
  public FileBackedGraffitiProvider(final Optional<Bytes32> defaultGraffiti) {
    this(defaultGraffiti, Optional.empty());
  }

  public FileBackedGraffitiProvider(final Optional<Path> graffitiFile) {
    this(Optional.empty(), graffitiFile);
  }
```

so I explored adding

```
  public FileBackedGraffitiProvider(final Bytes32 defaultGraffiti) {
    this(Optional.ofNullable(defaultGraffiti), Optional.empty());
  }

  public FileBackedGraffitiProvider(final Path graffitiFile) {
    this(Optional.empty(), Optional.ofNullable(graffitiFile));
  }
```

but as both of these constructors would only be used by test code I chose not to implement them.